simulator/host_system.go

```diff
@@ -173,6 +173,26 @@ func CreateStandaloneHost(f *Folder, spec types.HostConnectSpec) (*HostSystem, t
 	return host, nil
 }
 
+func (h *HostSystem) DestroyTask(req *types.Destroy_Task) soap.HasFault {
+	task := CreateTask(h, "destroy", func(t *Task) (types.AnyType, types.BaseMethodFault) {
+		for _, vmref := range h.Vm {
+			vm := Map.Get(vmref).(*VirtualMachine)
+			vm.Runtime.ConnectionState = types.VirtualMachineConnectionStateDisconnected
```

Comment:
Don't want to change ConnectionState here "When the API is provided directly by a server product, such as ESX Server, then the disconnected state is not possible", see: http://pubs.vmware.com/vsphere-6-5/index.jsp#com.vmware.wssdk.apiref.doc/vim.VirtualMachine.ConnectionState.html

Also note that with the new locking mechanism, the HostSystem (`h`) will be locked here, but VirtualMachine will not.  So you'd want to use `Map.WithLock(vm, func() {...})`

I think the task should return an error such as `ResourceInUse` if `len(h.Vm) != 0`, like what happens against real VC.
A host can't be destroyed unless VMs have been migrated elsewhere (or VMs have been deleted).  There's probably other checks that real VC makes, but again I think it's fine to leave those checks out for now.

Reply:
AH. I see. Thanks for pointing out the correct behavior.